### PR TITLE
Add geoshape query documentation

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add geoshape query documentation

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VijayanB — APPROVED** [review]

**Comment by VijayanB** [comment]
LGTM

File: _query-dsl/geo-and-xy/geoshape.md [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+        "_source": {
+          "location": "41.5623, 72.6506"
+        }
+      }
+    ]
+  }
+}
+```
+
+Note that when you indexed the geopoints, you specified their coordinates in the  `"latitude, longitude"` format. When you search for matching documents, the coordinate array is in the `[longitude, latitude]` format. Thus, document 1 is returned in the results but document 2 is not.
```
Comment (hdhalter) [inline _query-dsl/geo-and-xy/geoshape.md:722]:
```suggestion
Note that when you indexed the geopoints, you specified their coordinates in the `"latitude, longitude"` format. When you search for matching documents, the coordinate array is in the `[longitude, latitude]` format. Thus, document 1 is returned in the results but document 2 is not.
```

**Review by hdhalter — APPROVED** [review]
Looks really good! Just one nit.

File: _query-dsl/geo-and-xy/geoshape.md [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+
+Geoshape queries on geopoints only support the default `INTERSECTS` spatial relation, so you don't need to provide the `relation` parameter.
+{: .note}
+
+{: .important }
+> Geoshape queries on geopoints do not support the following geoshapes:
+> 
+> - Point
+> - Linestring
+> - Multipoint
```
Comment (natebower) [inline _query-dsl/geo-and-xy/geoshape.md:623]:
```suggestion
> - Multipoints
```

File: _query-dsl/geo-and-xy/geoshape.md [hunk 3]
```diff
@@ -0,0 +1,730 @@
...
+> - Multilinestring
```
Comment (natebower) [inline _query-dsl/geo-and-xy/geoshape.md:624]:
```suggestion
> - Multilinestrings
```

File: _query-dsl/geo-and-xy/geoshape.md [hunk 4]
```diff
@@ -0,0 +1,730 @@
...
+PUT /testindex1/_doc/2
+{
+  "location": "76.0254, 39.2467" 
+}
+```
+{% include copy-curl.html %}
+
+ For information about providing geopoint coordinates in various formats, see [Formats]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/geo-point/#formats). 
+
+Search for geopoints that intersect with the provided polygon:
```
Comment (natebower) [inline _query-dsl/geo-and-xy/geoshape.md:663]:
```suggestion
Search for geopoints that intersect the provided polygon:
```

File: _query-dsl/geo-and-xy/geoshape.md [hunk 5]
```diff
@@ -0,0 +1,730 @@
...
+        "_source": {
+          "location": "41.5623, 72.6506"
+        }
+      }
+    ]
+  }
+}
+```
+
+Note that when you indexed the geopoints, you specified their coordinates in the `"latitude, longitude"` format. When you search for matching documents, the coordinate array is in the `[longitude, latitude]` format. Thus, document 1 is returned in the results but document 2 is not.
```
Comment (natebower) [inline _query-dsl/geo-and-xy/geoshape.md:722]:
```suggestion
Note that when you indexed the geopoints, you specified their coordinates in `"latitude, longitude"` format. When you search for matching documents, the coordinate array is in `[longitude, latitude]` format. Thus, document 1 is returned in the results but document 2 is not.
```

File: _query-dsl/geo-and-xy/geoshape.md [hunk 6]
```diff
@@ -0,0 +1,730 @@
...
+
+## Request fields
+
+Geoshape queries accept the following fields.
+
+Field | Data type | Description
+:--- | :--- | :--- 
+`ignore_unmapped` | Boolean | Specifies whether to ignore an unmapped field. If set to `true`, then the query does not return any documents that contain an unmapped field. If set to `false`, then an exception is thrown when the field is unmapped. Optional. Default is `false`.
```
Comment (natebower) [inline _query-dsl/geo-and-xy/geoshape.md:730]:
```suggestion
`ignore_unmapped` | Boolean | Specifies whether to ignore an unmapped field. If set to `true`, then the query does not return any documents that contain an unmapped field. If set to `false`, then an exception is thrown when a field is unmapped. Optional. Default is `false`.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _query-dsl/geo-and-xy/geoshape.md [hunk 7]
```diff
@@ -0,0 +1,730 @@
...
+The searched document field must be mapped as `geo_point` or `geo_shape`.
+{: .note}
+
+## Spatial relations
+
+When you provide a geoshape to the geoshape query, the geopoint and geoshape fields in the documents are matched using the following spatial relations to the provided shape.
+
+Relation | Description | Supporting geographic field type
+:--- | :--- | :--- 
+`INTERSECTS` | (Default) Matches documents whose geopoint or geoshape intersects the shape provided in the query. | `geo_point`, `geo_shape`
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _query-dsl/geo-and-xy/geoshape.md:21]: Adding "with" for consistency here. I prefer to use "intersects with" throughout.

File: _query-dsl/geo-and-xy/geoshape.md [hunk 8]
```diff
@@ -0,0 +1,730 @@
...
+    ]
+  }
+}
+```
+
+## Querying geopoints
+
+You can also use a geoshape query to search for documents that contain geopoints. 
+
+Geoshape queries on geopoints only support the default `INTERSECTS` spatial relation, so you don't need to provide the `relation` parameter.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _query-dsl/geo-and-xy/geoshape.md:615]: I can change to "on geopoint fields". With or containing is not the right meaning 😄

Comment (kolchfa-aws) [inline _query-dsl/geo-and-xy/geoshape.md:615]:
```suggestion
Geoshape queries on geopoint fields only support the default `INTERSECTS` spatial relation, so you don't need to provide the `relation` parameter.
```

File: _query-dsl/geo-and-xy/geoshape.md [hunk 9]
```diff
@@ -0,0 +1,730 @@
...
+{: .note}
+
+{: .important }
+> Geoshape queries on geopoints do not support the following geoshapes:
```
Comment (kolchfa-aws) [inline _query-dsl/geo-and-xy/geoshape.md:619]:
```suggestion
> Geoshape queries on geopoint fields do not support the following geoshapes:
```

File: _query-dsl/geo-and-xy/geoshape.md [hunk 10]
```diff
@@ -0,0 +1,730 @@
...
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+For more information about specifying a different vertex orientation for polygons, see [Polygon]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/geo-shape/#polygon). 
+
+Index a polygon that specifies the search boundaries into the `pre-indexed-shapes` index. The polygon's ID is `search_triangle`. In this example, you'll index the polygon in WKT format:
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _query-dsl/geo-and-xy/geoshape.md:461]: Reworded.

Comment (kolchfa-aws) [inline _query-dsl/geo-and-xy/geoshape.md:461]:
```suggestion
Index a polygon specifying the search boundaries into the `pre-indexed-shapes` index. The polygon's ID is `search_triangle`. In this example, you'll index the polygon in WKT format:
```

File: _query-dsl/geo-and-xy/geoshape.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _query-dsl/geo-and-xy/geoshape.md:615]: That's much clearer, thanks :-)

**Pull request merged**